### PR TITLE
FUSETOOLS-1826 : invoke the cleanMarkers method

### PR DESCRIPTION
**Pull request by vcornejo**
Invoke cleanMarkers method to remove all validation messages associated with the deleted camel endpoint.

### CODE REVIEW

**Comment by apupier** [comment]
testPR

**Comment by vcornejo** [comment]
Ok I agree, on preliminar test the `eventBroker.post(AbstractCamelModelElement.TOPIC_REMOVE_CAMEL_ELEMENT,cme); ` works fine, but I need clarify the right place where call the post : 
CamelGlobalConfigEditor.deleteEntries() or GlobalEndpointContributor.onGlobalElementDeleted ().

**Comment by apupier** [comment]
I think it should be handled directly in the model. So in CamelContextElement.removeEndpointDefinition and CamelContextElement.removeDataFormat and CamelFile.removeGlobalDefinition

This kind of modification can be highly impacting so we will need to write tests.
I think that the plugin in which it will make more sense to add it is org.fusesource.ide.camel.validation.tests.integration
You can get some inspiration to write the test from org.fusesource.ide.camel.editor.integration.CamelEditorIT in order to import a camel file in a project with some validation errors and then try to call the delete entries from the GlobalConfigEditor.
If too complicated, we may ask to QE team to write a UI test.

**Comment by vcornejo** [comment]
Ok, my UI test going well (inspired on CamelEditorIT ), only have 2 comments :
- To get the Markers from the ProblemsView I use reflection to invoke getAllMarkers()
- The last problem that I have is how to simulate click on delete button? any suggestion

Regards.

**Comment by apupier** [comment]
@vcornejo I see no changes in the PR, can you check the status of your branch please?

**Comment by vcornejo** [comment]
I was committed the changes to the branch. Can you see the changes?

File: core/tests/org.fusesource.ide.camel.validation.tests.integration/src/org/fusesource/ide/camel/validator/marker/MarkersRemoverValidator.java [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+import org.fusesource.ide.camel.editor.globalconfiguration.CamelGlobalConfigEditor;
+import org.fusesource.ide.camel.model.service.core.model.AbstractCamelModelElement;
+import org.fusesource.ide.camel.model.service.core.tests.integration.core.io.FuseProject;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+
+
+public class MarkersRemoverValidator {
```
Comment (apupier) [inline core/tests/org.fusesource.ide.camel.validation.tests.integration/src/org/fusesource/ide/camel/validator/marker/MarkersRemoverValidator.java]: test name should end with IT

File: core/tests/org.fusesource.ide.camel.validation.tests.integration/META-INF/MANIFEST.MF [hunk 2]
```diff
@@ -14,5 +14,13 @@ Require-Bundle: org.assertj.core;bundle-version="2.1.0",
...
- org.fusesource.ide.camel.model.service.v2_17_3;bundle-version="9.1.0"
+ org.fusesource.ide.camel.model.service.v2_17_3;bundle-version="9.1.0",
+ org.fusesource.ide.camel.model.service.core.tests.integration,
+ org.eclipse.emf.common;bundle-version="2.11.0",
+ org.eclipse.graphiti;bundle-version="0.13.0",
+ org.eclipse.emf.transaction;bundle-version="1.8.0",
+ org.fusesource.ide.branding;bundle-version="9.1.0",
+ org.fusesource.ide.camel.editor;bundle-version="9.1.0",
+ org.eclipse.ui.ide;bundle-version="3.12.0",
+ org.eclipse.text
```
Comment (apupier) [inline core/tests/org.fusesource.ide.camel.validation.tests.integration/META-INF/MANIFEST.MF]: this dependency seems strange, are you sure that it is needed?

File: core/tests/org.fusesource.ide.camel.validation.tests.integration/src/org/fusesource/ide/camel/validator/marker/MarkersRemoverValidator.java [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+		statusHandlerCalled = false;
+		openEditorOnFileStore = openFileInEditorWithProblemsViewOpened("/route.xml");
+		Policy.setStatusHandler(new StatusHandler() {
+			
+			@Override
+			public void show(IStatus status, String title) {
+				statusHandlerCalled = true;
+			}
+		});
+
```
Comment (apupier) [inline core/tests/org.fusesource.ide.camel.validation.tests.integration/src/org/fusesource/ide/camel/validator/marker/MarkersRemoverValidator.java]: there are 3 useless empty line

File: core/tests/org.fusesource.ide.camel.validation.tests.integration/src/org/fusesource/ide/camel/validator/marker/MarkersRemoverValidator.java [hunk 4]
```diff
@@ -0,0 +1,205 @@
...
+		
+	}
+	
+	@After
+	public void tearDown() throws Exception {
+		IWorkbenchPage page = PlatformUI.getWorkbench().getActiveWorkbenchWindow().getActivePage();
+		page.closeAllEditors(false);
+		SafeRunnable.setIgnoreErrors(safeRunnableIgnoreErrorStateBeforeTests);
+		Policy.setStatusHandler(statusHandlerBeforetest);
+
```
Comment (apupier) [inline core/tests/org.fusesource.ide.camel.validation.tests.integration/src/org/fusesource/ide/camel/validator/marker/MarkersRemoverValidator.java]: there is one useless empty line

File: core/tests/org.fusesource.ide.camel.validation.tests.integration/src/org/fusesource/ide/camel/validator/marker/MarkersRemoverValidator.java [hunk 5]
```diff
@@ -0,0 +1,205 @@
...
+	@After
+	public void tearDown() throws Exception {
+		IWorkbenchPage page = PlatformUI.getWorkbench().getActiveWorkbenchWindow().getActivePage();
+		page.closeAllEditors(false);
+		SafeRunnable.setIgnoreErrors(safeRunnableIgnoreErrorStateBeforeTests);
+		Policy.setStatusHandler(statusHandlerBeforetest);
+		
+	}
+	
+
```
Comment (apupier) [inline core/tests/org.fusesource.ide.camel.validation.tests.integration/src/org/fusesource/ide/camel/validator/marker/MarkersRemoverValidator.java]: there are 2 useless empty lines

File: core/tests/org.fusesource.ide.camel.validation.tests.integration/src/org/fusesource/ide/camel/validator/marker/MarkersRemoverValidator.java [hunk 6]
```diff
@@ -0,0 +1,205 @@
...
+        Button deleteButton= (Button) deleteButtonField.get(globalEditor);
+        
+       
+        for(int i=0;i<elements.size();i++){
+        	//select element
+           globalEditor.setSelection((AbstractCamelModelElement)elements.get(i));
+            //delet selected element from the tree       
+           deleteButton.notifyListeners(SWT.Selection,null);
+           readAndDispatch(20);
+
```
Comment (apupier) [inline core/tests/org.fusesource.ide.camel.validation.tests.integration/src/org/fusesource/ide/camel/validator/marker/MarkersRemoverValidator.java]: there are 2 useless empty lines

File: core/tests/org.fusesource.ide.camel.validation.tests.integration/src/org/fusesource/ide/camel/validator/marker/MarkersRemoverValidator.java [hunk 7]
```diff
@@ -0,0 +1,205 @@
...
+       
+        for(int i=0;i<elements.size();i++){
+        	//select element
+           globalEditor.setSelection((AbstractCamelModelElement)elements.get(i));
+            //delet selected element from the tree       
+           deleteButton.notifyListeners(SWT.Selection,null);
+           readAndDispatch(20);
+           
+			
+		}
```
Comment (apupier) [inline core/tests/org.fusesource.ide.camel.validation.tests.integration/src/org/fusesource/ide/camel/validator/marker/MarkersRemoverValidator.java]: it seems that this bracket is not correctly indented

File: core/tests/org.fusesource.ide.camel.validation.tests.integration/src/org/fusesource/ide/camel/validator/marker/MarkersRemoverValidator.java [hunk 8]
```diff
@@ -0,0 +1,205 @@
...
+        
+        globalEditor.doSave(null);
+        
+        Thread.currentThread().sleep(3000);
+        readAndDispatch(20);
+        readAndDispatch(20);
+       
+        markers = (IMarker[]) getAllMarkersMethod.invoke(problemView, new Object[]{});
+        assertThat(markers.length==0);
+
```
Comment (apupier) [inline core/tests/org.fusesource.ide.camel.validation.tests.integration/src/org/fusesource/ide/camel/validator/marker/MarkersRemoverValidator.java]: there are 3 useless empty lines

File: core/tests/org.fusesource.ide.camel.validation.tests.integration/src/org/fusesource/ide/camel/validator/marker/MarkersRemoverValidator.java [hunk 9]
```diff
@@ -0,0 +1,205 @@
...
+		
+		Thread.currentThread().sleep(1000);
+		
+		readAndDispatch(20);
+		
+		      
+        Method getAllMarkersMethod = ExtendedMarkersView.class.getDeclaredMethod("getAllMarkers", new Class[]{}); //$NON-NLS-1$
+		getAllMarkersMethod.setAccessible(true);
+		IMarker[] markers = (IMarker[]) getAllMarkersMethod.invoke(problemView, new Object[]{});
```
Comment (apupier) [inline core/tests/org.fusesource.ide.camel.validation.tests.integration/src/org/fusesource/ide/camel/validator/marker/MarkersRemoverValidator.java]: indentation issue

File: core/tests/org.fusesource.ide.camel.validation.tests.integration/src/org/fusesource/ide/camel/validator/marker/MarkersRemoverValidator.java [hunk 10]
```diff
@@ -0,0 +1,205 @@
...
+		}
+
+		
+		Thread.currentThread().sleep(1000);
+		
+		readAndDispatch(20);
+		
+		      
+        Method getAllMarkersMethod = ExtendedMarkersView.class.getDeclaredMethod("getAllMarkers", new Class[]{}); //$NON-NLS-1$
+		getAllMarkersMethod.setAccessible(true);
```
Comment (apupier) [inline core/tests/org.fusesource.ide.camel.validation.tests.integration/src/org/fusesource/ide/camel/validator/marker/MarkersRemoverValidator.java]: indentation issue

File: core/tests/org.fusesource.ide.camel.validation.tests.integration/src/org/fusesource/ide/camel/validator/marker/MarkersRemoverValidator.java [hunk 11]
```diff
@@ -0,0 +1,205 @@
...
+		IWorkbenchPart welcomePage = PlatformUI.getWorkbench().getActiveWorkbenchWindow().getActivePage().getActivePart();
+		welcomePage.dispose();
+		page.closeAllEditors(false);
+		page.closeAllPerspectives(false, false);
+		PlatformUI.getWorkbench().showPerspective(FusePerspective.ID, page.getWorkbenchWindow());
+		safeRunnableIgnoreErrorStateBeforeTests = SafeRunnable.getIgnoreErrors();
+		SafeRunnable.setIgnoreErrors(false);
+		statusHandlerBeforetest = Policy.getStatusHandler();
+		statusHandlerCalled = false;
+		openEditorOnFileStore = openFileInEditorWithProblemsViewOpened("/route.xml");
```
Comment (apupier) [inline core/tests/org.fusesource.ide.camel.validation.tests.integration/src/org/fusesource/ide/camel/validator/marker/MarkersRemoverValidator.java]: I suppose that you forget to commit the route.xml file? Or I miss something?

File: core/tests/org.fusesource.ide.camel.validation.tests.integration/META-INF/MANIFEST.MF (hunk 2 above)
Comment (vcornejo) [inline core/tests/org.fusesource.ide.camel.validation.tests.integration/META-INF/MANIFEST.MF]: These 2 aren't needed
+ org.eclipse.ui.ide;bundle-version="3.12.0",
+ org.eclipse.text

File: core/tests/org.fusesource.ide.camel.validation.tests.integration/src/org/fusesource/ide/camel/validator/marker/MarkersRemoverValidator.java (hunk 8 above)
Comment (vcornejo) [inline core/tests/org.fusesource.ide.camel.validation.tests.integration/src/org/fusesource/ide/camel/validator/marker/MarkersRemoverValidator.java]: Fixed

File: core/tests/org.fusesource.ide.camel.validation.tests.integration/src/org/fusesource/ide/camel/validator/marker/MarkersRemoverValidator.java (hunk 7 above)
Comment (vcornejo) [inline core/tests/org.fusesource.ide.camel.validation.tests.integration/src/org/fusesource/ide/camel/validator/marker/MarkersRemoverValidator.java]: Fixed

File: core/tests/org.fusesource.ide.camel.validation.tests.integration/src/org/fusesource/ide/camel/validator/marker/MarkersRemoverValidator.java (hunk 9 above)
Comment (vcornejo) [inline core/tests/org.fusesource.ide.camel.validation.tests.integration/src/org/fusesource/ide/camel/validator/marker/MarkersRemoverValidator.java]: Fixed

File: core/tests/org.fusesource.ide.camel.validation.tests.integration/src/org/fusesource/ide/camel/validator/marker/MarkersRemoverValidator.java (hunk 5 above)
Comment (vcornejo) [inline core/tests/org.fusesource.ide.camel.validation.tests.integration/src/org/fusesource/ide/camel/validator/marker/MarkersRemoverValidator.java]: Fixed

File: core/tests/org.fusesource.ide.camel.validation.tests.integration/src/org/fusesource/ide/camel/validator/marker/MarkersRemoverValidator.java (hunk 4 above)
Comment (vcornejo) [inline core/tests/org.fusesource.ide.camel.validation.tests.integration/src/org/fusesource/ide/camel/validator/marker/MarkersRemoverValidator.java]: Fixed

File: core/tests/org.fusesource.ide.camel.validation.tests.integration/src/org/fusesource/ide/camel/validator/marker/MarkersRemoverValidator.java (hunk 11 above)
Comment (vcornejo) [inline core/tests/org.fusesource.ide.camel.validation.tests.integration/src/org/fusesource/ide/camel/validator/marker/MarkersRemoverValidator.java]: Ups, fixed

**Comment by apupier** [comment]
testPR

File: core/tests/org.fusesource.ide.camel.validation.tests.integration/.classpath [hunk 12]
```diff
@@ -1,7 +1,8 @@
-<?xml version="1.0" encoding="UTF-8"?>
-<classpath>
-	<classpathentry kind="con" path="org.eclipse.jdt.launching.JRE_CONTAINER/org.eclipse.jdt.internal.debug.ui.launcher.StandardVMType/JavaSE-1.8"/>
-	<classpathentry kind="con" path="org.eclipse.pde.core.requiredPlugins"/>
-	<classpathentry kind="src" path="src"/>
-	<classpathentry kind="output" path="bin"/>
-</classpath>
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (apupier) [inline core/tests/org.fusesource.ide.camel.validation.tests.integration/.classpath]: seems that only line-ending has been modified. Please not include this file in this PR

File: core/plugins/org.fusesource.ide.camel.model.service.core/src/org/fusesource/ide/camel/model/service/core/model/AbstractCamelModelElement.java [hunk 13]
```diff
@@ -1754,4 +1753,16 @@ public Element createElement(String nodeTypeId, String namespace) {
 	public String getTranslatedNodeName() {
 		return CamelUtils.getTranslatedNodeName(getXmlNode());
 	}
+	
+	/**
+	 * Notify listeners after remove Context Element
```
Comment (apupier) [inline core/plugins/org.fusesource.ide.camel.model.service.core/src/org/fusesource/ide/camel/model/service/core/model/AbstractCamelModelElement.java]: it is not only about ContextElement, it is about removal of the modelElement which is passed in parameter

**Comment by lhein** [comment]
testPR

**Comment by apupier** [comment]
testPR

**Comment by apupier** [comment]
Please rebase your PR.
The conflict might be only related to some Line ending issue. We are now using Linux style on all files and enforcing it.
Sorry for the inconvenience

**Comment by apupier** [comment]
@vcornejo it seems that you have some troubles with the line-ending, I created a branch in which the line-ending should be fixed: https://github.com/apupier/fuseide/tree/vcornejo-FUSETOOLS-1826
I think that the easiest is that you reset your branch on mine.

For the development of another feature I had to create an AbstractCamelEditorIT. You should try to inherit from it. Maybe need to move it to another common plugin.

**Pull request closed without merging**